File: problems/String to Integer (atoi)/myAtoi.cs
public class Solution {
    public int MyAtoi(string str) {
        str = str.Trim();

        if (0 == str.Length) {
            return 0;
        }

        var negative = '-' == str[0];
        var positive = '+' == str[0];
        var i = negative || positive ? 1 : 0;
        long result = 0;

        while (str.Length > i && char.IsDigit(str[i])) {
            result = 10 * result + (str[i++] - '0');
            if (result > int.MaxValue) {
                return negative ? int.MinValue : int.MaxValue;
            }
        }

        return negative ? -(int)result : (int)result;
    }
}

File: problems/String to Integer (atoi)/myAtoi.java
class Solution {
    public int myAtoi(String s) {
        s = s.trim();

        int n = s.length();

        if (0 == n) {
            return 0;
        }

        boolean negative = '-' == s.charAt(0);
        boolean positive = '+' == s.charAt(0);
        int idx = negative || positive ? 1 : 0;
        long result = 0;

        while (n > idx && Character.isDigit(s.charAt(idx))) {
            result = 10 * result + (s.charAt(idx++) - '0');

            if (result > Integer.MAX_VALUE) {
                return negative ? Integer.MIN_VALUE : Integer.MAX_VALUE;
            }
        }

        return negative ? -(int)result : (int)result;
    }
}
